core/tee/tee_rpmb_fs.c

```diff
@@ -1264,18 +1272,111 @@ static TEE_Result tee_rpmb_read(uint16_t dev_id, uint32_t addr, uint8_t *data,
 	return res;
 }
 
+static TEE_Result write_req(uint16_t dev_id, uint16_t blk_idx,
+			    const void *data_blks, uint16_t blkcnt,
+			    const uint8_t *fek, const TEE_UUID *uuid,
+			    struct tee_rpmb_mem *mem, void  *req, void *resp)
+{
+	TEE_Result res = TEE_SUCCESS;
+	uint8_t hmac[RPMB_KEY_MAC_SIZE] = { };
+	uint32_t wr_cnt = rpmb_ctx->wr_cnt;
+	struct rpmb_raw_data rawdata = { };
+	size_t retry_count = 0;
+
+	assert(mem->req_size <=
+	       sizeof(struct rpmb_req) + blkcnt * RPMB_DATA_FRAME_SIZE);
+	assert(mem->resp_size <= RPMB_DATA_FRAME_SIZE);
+
+	while (true) {
+		memset(req, 0, mem->req_size);
+		memset(resp, 0, mem->resp_size);
+
+		memset(&rawdata, 0, sizeof(struct rpmb_raw_data));
+		rawdata.msg_type = RPMB_MSG_TYPE_REQ_AUTH_DATA_WRITE;
+		rawdata.block_count = &blkcnt;
+		rawdata.blk_idx = &blk_idx;
+		rawdata.write_counter = &wr_cnt;
+		rawdata.key_mac = hmac;
+		rawdata.data = (uint8_t *)data_blks;
+
+		res = tee_rpmb_req_pack(req, &rawdata, blkcnt, dev_id, fek,
+					uuid);
+		if (res) {
```

Comment:
Should this be if (res != TEE_SUCCESS) to match the test for tee_rpmb_invoke below?

Reply:
You're right, but too late I'm afraid.

Reply:
Yeah, `if (res)` doesn't particularly stand out during review because it is now the recommended coding style. Older files tend to have `if (res != TEE_SUCCESS)` more often than not, which raises the question what to do when updating those files. My preference is to use `if (res != TEE_SUCESS)` when modifying a function where this syntax is used already (as you said, for consistency), but use `if (res)` in new functions.